src/Data/Foreign.purs

```diff
@@ -44,16 +44,20 @@ foreign import data Foreign :: *
 
 -- | A type for runtime type errors
 data ForeignError
-  = TypeMismatch String String
+  = TypeMismatch (Array String) String
   | ErrorAtIndex Int ForeignError
   | ErrorAtProperty String ForeignError
   | JSONError String
 
 instance showForeignError :: Show ForeignError where
```

Comment:
While we're breaking things here, we should make this into a valid `Show` instance, and move this into a `renderForeignError` function.